modules/percolator/src/main/java/org/elasticsearch/percolator/TransportPercolateAction.java

```diff
@@ -184,6 +184,9 @@ public static SearchRequest createSearchRequest(PercolateRequest percolateReques
                         }
                     } else if (token.isValue()) {
                         if ("size".equals(currentFieldName)) {
+                            if (percolateRequest.onlyCount()) {
```

Comment:
Is this a leftover or did you move it or something? It feels like it doesn't belong.